### PR TITLE
Add eck entity entity trait.

### PR DESCRIPTION
**Pull request by richardgaunt**
## Background

I've been using [Entity Contruction Kit](https://www.drupal.org/project/eck) to implement a custom content entity and realised that there was no way to test the entity with behat-steps.

This PR implements a ECK entity trait to test ECK entities. It should work on ECK entities.

## What has changed
1. Implemented eck entity trait
2. Added eck module to composer
3. Added test eck entity fixture to config
2. Added tests to verify the trait's functionality.

### CODE REVIEW

**Comment by AlexSkrypnyk** [comment]
This PR requires:
1. Addition of ECK module to the D7 fixture
2. Adding a test feature that calls step definitions provided by the trait in this PR

**Comment by richardgaunt** [comment]
@AlexSkrypnyk This is a PR for the eck module in Drupal8. It's ready for review.

File: src/D8/EckEntityTrait.php [hunk 1]
```diff
@@ -0,0 +1,196 @@
+<?php
```
Comment (AlexSkrypnyk) [inline src/D8/EckEntityTrait.php:1]: the name of the trait is usually a module name.

Can this please be changed to `EckTrait.php`

File: src/D8/EckEntityTrait.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+   * Create eck entities.
+   *
+   * Provide entity data in the following format:
+   * | title  | field_marine_animal     | field_fish_type | ... |
+   * | Snook  | Fish                    | Marine fish     | 10  |
+   * | ...    | ...                     | ...         | ...     |
+   *
+   * @Given :bundle :entity_type entities:
+   */
+  public function eckEntitiesCreate($bundle, $entity_type, TableNode $table) {
```
Comment (AlexSkrypnyk) [inline src/D8/EckEntityTrait.php:33]: please update all method names to start with `eck` as it is used as a namespace.

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@richardgaunt 
Thank you for your contribution! This looks amazing!

Could you please update the file name and methods to start with `eck`. 

Thank you

File: src/D8/EckEntityTrait.php (hunk 1 above)
Comment (richardgaunt) [reply src/D8/EckEntityTrait.php]: @AlexSkrypnyk done thanks for the review

**Review by AlexSkrypnyk — APPROVED** [review]

**Pull request merged**